flink-contrib/flink-timerserivce-rocksdb/src/main/java/org/apache/flink/contrib/streaming/api/operators/RocksDBInternalTimerService.java

```diff
@@ -0,0 +1,797 @@
+				fileSystem.delete(this.dbPath, true);
+			}
+			
+			fileSystem.mkdirs(dbPath);
+		} catch (IOException e) {
+			throw new RuntimeException("Error while creating directory for rocksdb timer service.", e);
+		}
+
+		ColumnFamilyOptions columnFamilyOptions = new ColumnFamilyOptions()
+				.setMergeOperator(new StringAppendOperator())
+				.setCompactionStyle(CompactionStyle.UNIVERSAL);
+		ColumnFamilyDescriptor defaultColumnDescriptor = new ColumnFamilyDescriptor("default".getBytes(), columnFamilyOptions);
+
+		DBOptions dbOptions = new DBOptions()
+				.setCreateIfMissing(true)
+				.setUseFsync(false)
+				.setDisableDataSync(true)
+				.setMaxOpenFiles(-1);
+
+		List<ColumnFamilyHandle> columnFamilyHandles = new ArrayList<>(1);
+
+		try {
+			this.db = RocksDB.open(dbOptions, dbPath.getPath(), Collections.singletonList(defaultColumnDescriptor), columnFamilyHandles);
+		} catch (RocksDBException e) {
+			throw new RuntimeException("Error while creating the RocksDB instance.", e);
+		}
+
+		this.numPartitions = Math.min(keyGroupRange.getNumberOfKeyGroups(), MAX_PARTITIONS);
+
+		ColumnFamilyHandle eventTimeColumnFamilyHandle;
+		ColumnFamilyHandle processingTimeColumnFamilyHandle;
+		try {
+			ColumnFamilyDescriptor eventTimeColumnFamilyDescriptor = new ColumnFamilyDescriptor("eventTime".getBytes(), columnFamilyOptions);
+			ColumnFamilyDescriptor processingTimeColumnFamilyDescriptor = new ColumnFamilyDescriptor("processingTime".getBytes(), columnFamilyOptions);
+			eventTimeColumnFamilyHandle = db.createColumnFamily(eventTimeColumnFamilyDescriptor);
+			processingTimeColumnFamilyHandle = db.createColumnFamily(processingTimeColumnFamilyDescriptor);
+		} catch (RocksDBException e) {
+			throw new RuntimeException("Error while creating the column families.", e);
+		}
+
+		this.processingTimeHeap = new PersistentTimerHeap(numPartitions, processingTimeColumnFamilyHandle);
+		this.eventTimeHeap = new PersistentTimerHeap(numPartitions, eventTimeColumnFamilyHandle);
+	}
+
+	// ------------------------------------------------------------------------
+	//  InternalTimerService Implementation
+	// ------------------------------------------------------------------------
+	
+	@Override
+	public void start() {
+		// rebuild the heaps
+		eventTimeHeap.initialize();
+		processingTimeHeap.initialize();
+		
+		// register the processing timer with the minimum timestamp
+		Tuple4<Integer, Long, K, N> headProcessingTimer = processingTimeHeap.top();
+		if (headProcessingTimer != null) {
+			nextTimer = processingTimeService.registerTimer(headProcessingTimer.f1, this);
+		}
+	}
+
+	@Override
+	public void close() {
+		if (db != null) {
+			db.close();
+		}
+		
+		if (dbPath != null) {
+			try {
+				FileSystem fileSystem = dbPath.getFileSystem();
+				if (fileSystem.exists(dbPath)) {
+					fileSystem.delete(dbPath, true);
+				}
+			} catch (IOException e) {
+				throw new RuntimeException("Error while cleaning directory for rocksdb timer service.", e);
```

Comment:
i think it is important that this can somehow report IOException. We might need a way to react somehow in the calling code in case the delete went wrong, to prevent lingering files. Se also my comment about adding the exception to the superclass's method.

Reply:
I'd suggest creating a `FlinkRocksDBException` to wrap `RocksDBException` and throw it